### ISSUE TITLE
Implement more fuzzy matching for format mismatch in URL QA

### ISSUE THREAD

**Issue by philipashlock** [GSA/project-open-data-dashboard#75]
The format mismatch in the downloadURL/accessURL QA is mostly designed to detect when resources are linking to webpages instead of the expected file format, but right now it simply reports whether there's an exact match between the media type specified in the data.json metadata and the `Content-Type` returned in the HTTP header for the given URL. This check should allow fuzzier matches so that we record warnings for a mismatch between mimetypes that could be considered synonymous, eg `application/zip` and `application/x-zip-compressed`

In other words if there's a mismatch between something like `application/csv` and `text/html` that should probably return an error whereas if the mismatch is between `application/zip` and `application/x-zip-compressed` then that should probably just be a warning. 

The distinction between warnings and errors was raised in #73

**Comment by dsmorgan77** [comment]
This is great. I think where you now is basically testing the difference between the schema requirements for downloadURL and accessURL:

* If someone provides a downloadURL that points to something that’s actually returning text/html, then it really ought to be marked accessURL.
* If someone provides an accessURL that points to something that’s actually returning application/csv, then it really ought to be marked downloadURL

In those cases, that's a straight-up schema compliance issue and should be throwing an error.
